cmd/ebrelayer/relayer/prophecyHandler.go

```diff
@@ -0,0 +1,82 @@
+package relayer
+
+import (
+	cosmosbridge "github.com/Sifchain/sifnode/cmd/ebrelayer/contract/generated/bindings/cosmosbridge"
+	"github.com/Sifchain/sifnode/cmd/ebrelayer/txs"
+	"github.com/Sifchain/sifnode/cmd/ebrelayer/types"
+)
+
+var currentGlobalNonce = uint64(0)
+var prophecyInfoBuffer = map[uint64]types.ProphecyInfo{}
+
+// Parses event data from the msg, event, builds a new ProphecyClaim, and relays it to Ethereum
+func (sub CosmosSub) handleProphecyCompleted(
+	prophecyInfo types.ProphecyInfo,
+) {
+	sub.SugaredLogger.Infow(
+		"get the prophecy completed message.",
+		"cosmosMsg", prophecyInfo,
+	)
+
+	// will discard it if global nonce is less than current global nonce
+	if currentGlobalNonce > prophecyInfo.GlobalNonce {
+		sub.SugaredLogger.Errorw("global nonce is invalid.",
+			"current global nonce", currentGlobalNonce,
+			"global nonce in message", prophecyInfo.GlobalNonce)
+		return
+	}
+
+	// buffer the prophecy
+	if prophecyInfo.GlobalNonce > currentGlobalNonce+1 {
+		prophecyInfoBuffer[prophecyInfo.GlobalNonce] = prophecyInfo
```

Comment:
Is this `prophecyInfoBuffer` ever used? I only see two references to it, once on creation and second here where things are added to it. I never see the buffered items read.

Reply:
will remove it